lib/jobs/dell-wsman-add-hotspare-updatexml.js

```diff
@@ -0,0 +1,171 @@
+            self._done(err);
+        });
+    };
+
+    WsmanAddHotspareUpdateXmlJob.prototype.updateXml = function(data) {
+        var self = this;
+        var xmlFile = String.fromCharCode.apply(null, new Uint16Array(data));
+        var doc = new xmldom().parseFromString(xmlFile, 'application/xml');
+        if(self.options.hotspareType === 'dhs') {
+            self.updateVolumeElement(doc);
+        } else {
+            self.updateDiskElement(doc);
+        }
+        return doc;
+    };
+
+    /*
+     * Dedicated hotspare should be added into corresponding volume
+     */
+    WsmanAddHotspareUpdateXmlJob.prototype.updateVolumeElement = function(doc) {
+        var self = this;
+        if(!self.options.volumeId) {
+            throw new errors.NotFoundError('The volumeId should not be empty.');
+        }
+        var components = doc.getElementsByTagName('Component');
+        _.forEach(components, function(component) {
+            var volumeFQDD = component.getAttribute('FQDD');
+            if(volumeFQDD === self.options.volumeId) {
+                var newNode = doc.createElement('Attribute');
+                newNode.setAttribute('Name', 'RAIDdedicatedSpare');
+                newNode.textContent = self.options.driveId;
+                component.appendChild(newNode);
+                var breakLineChild = doc.createTextNode('\n');
+                component.appendChild(breakLineChild);
+                return false;
+            }
+        });
+    };
+
+    /*
+     * Update hotspare status of specified drive
+     */
+    WsmanAddHotspareUpdateXmlJob.prototype.updateDiskElement = function(doc) {
+        var self = this;
+        var components = doc.getElementsByTagName('Component');
+        _.forEach(components, function(component) {
+            var diskFQDD = component.getAttribute('FQDD');
+            if(diskFQDD === self.options.driveId) {
+                //create a new drive node, to replace the original one
+                var newDriveNode = doc.createElement('Component');
+                newDriveNode.setAttribute('FQDD', diskFQDD);
+
+                //create children nodes for the new drive node
+                var statusChild = doc.createElement('Attribute');
+                statusChild.setAttribute('Name', 'RAIDHotSpareStatus');
+                statusChild.textContent = 'Global';
+                var stateChild = doc.createElement('Attribute');
+                stateChild.setAttribute('Name', 'RAIDPDState');
+                stateChild.textContent = 'Ready';
+
+                //create and insert break line child into new drive node,
+                //before inserting each attribute child
+                var breakLineChild = doc.createTextNode('\n');
+                newDriveNode.appendChild(breakLineChild);
+                newDriveNode.appendChild(statusChild);
+
+                breakLineChild = doc.createTextNode('\n');
```

Comment:
This line is duplicated with line 152

Reply:
Each node is identified as a instance, could not be appended to multiple nodes.